test/artillery/analysis.js

```diff
@@ -0,0 +1,131 @@
+const modes = require('./modes.js')
+const sampling = require('./sampling.js')
+
+const analysis = {
+  getErrorBudget: (script, defaultErrorBudget) => {
+    if (script && script.sampling && script.sampling.errorBudget !== undefined) {
+      return script.sampling.errorBudget
+    } else if (defaultErrorBudget !== undefined) {
+      return defaultErrorBudget
+    } else {
+      return sampling.defaults.sampling.errorBudget
+    }
+  },
+
+  /**
+   * Analyze a set of sampling reports, each of which is the result of a sample battery's execution
```

Comment:
I appreciate you adding documentation for the parameters and what each function does.  It really helps me figure out the logic of each line of code here! :)